### PR TITLE
More rhel9 fixes

### PR DESCRIPTION
**Pull request by matejak**
This PR

- enables more rules for RHEL9,
- fixes platform metadata for some RHEL9 rule remediations, and
- fixes the Grub2 config path in Fedora and RHEL9.
- adds the sshd directory config rule to respective profiles
- addresses a package merge

Fixes: #7282

### CODE REVIEW

File: linux_os/guide/system/accounts/accounts-physical/disable_ctrlaltdel_reboot/bash/shared.sh [hunk 1]
```diff
@@ -1,9 +1,5 @@
 # platform = Red Hat Virtualization 4,multi_platform_fedora,multi_platform_ol,multi_platform_rhel,multi_platform_wrlinux
 {{%- if init_system == "systemd" -%}}
-{{% if product in ["rhel7", "rhel8"] %}}
-# The process to disable ctrl+alt+del has changed in RHEL7. 
-# Reference: https://access.redhat.com/solutions/1123873
```
Comment (jan-cerny) [inline linux_os/guide/system/accounts/accounts-physical/disable_ctrlaltdel_reboot/bash/shared.sh:5]: Why is this removed?

Comment (matejak) [reply linux_os/guide/system/accounts/accounts-physical/disable_ctrlaltdel_reboot/bash/shared.sh:5]: As this project doesn't contain RHEL6 rules, we can now behave as if the remediation was like that from the beginning.

**Review by yuumasato — CHANGES_REQUESTED** [review]
LGTM, I just noticed a few typos.

File: products/rhel9/profiles/ospp.profile [hunk 2]
```diff
@@ -177,7 +178,6 @@ selections:
     - package_aide_installed
     - package_dnf-automatic_installed
     - package_subscription-manager_installed
-#    - package_dnf-plugin-subscription-manager_installed  # not supported in RHEL9 ATM
```
[the remark this answers is not in the document]
Comment (matejak) [reply products/rhel9/profiles/ospp.profile:180]: Yes, you can see the rationale in the commit, the rule is not needed in RHEL9.

[the remark this answers is not in the document]
Comment (yuumasato) [reply products/rhel9/profiles/ospp.profile:180]: Cool, I forgot to look into individual commits.

**Review by yuumasato — APPROVED** [review]
